File: codemap/tests/fixtures/sample_module.go
// Package sample provides sample Go code for testing.
package sample

import "fmt"

// User represents a user in the system.
type User struct {
	ID   int
	Name string
}

// UserService handles user operations.
type UserService interface {
	GetUser(id int) (*User, error)
	CreateUser(name string) (*User, error)
}

// DefaultService is the default implementation.
type DefaultService struct {
	users map[int]*User
}

// GetUser retrieves a user by ID.
func (s *DefaultService) GetUser(id int) (*User, error) {
	if user, ok := s.users[id]; ok {
		return user, nil
	}
	return nil, fmt.Errorf("user not found")
}

// CreateUser creates a new user.
func (s *DefaultService) CreateUser(name string) (*User, error) {
	user := &User{ID: len(s.users) + 1, Name: name}
	s.users[user.ID] = user
	return user, nil
}

// Helper function for greeting.
func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// Process handles async-like operations.
func Process(data []byte) ([]byte, error) {
	return data, nil
}

File: codemap/tests/fixtures/sample_module.rs
//! Sample Rust module for testing.

use std::error::Error;

/// Represents a user in the system.
pub struct User {
    pub id: u32,
    pub name: String,
}

/// Service trait for user operations.
pub trait UserService {
    fn get_user(&self, id: u32) -> Option<&User>;
    fn create_user(&mut self, name: String) -> &User;
}

/// User status enumeration.
pub enum UserStatus {
    Active,
    Inactive,
    Pending,
}

/// Implementation block for User.
impl User {
    /// Creates a new user.
    pub fn new(id: u32, name: String) -> Self {
        User { id, name }
    }

    /// Gets the display name.
    pub fn display_name(&self) -> String {
        format!("User: {}", self.name)
    }
}

/// Default service implementation.
pub struct DefaultService {
    users: Vec<User>,
}

impl UserService for DefaultService {
    fn get_user(&self, id: u32) -> Option<&User> {
        self.users.iter().find(|u| u.id == id)
    }

    fn create_user(&mut self, name: String) -> &User {
        let id = self.users.len() as u32 + 1;
        self.users.push(User::new(id, name));
        self.users.last().unwrap()
    }
}

/// Greets a user by name.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

/// Async function for processing data.
pub async fn process_data(data: Vec<u8>) -> Result<Vec<u8>, Box<dyn Error>> {
    Ok(data)
}

/// Submodule for utilities.
pub mod utils {
    /// Formats a number.
    pub fn format_number(n: u32) -> String {
        n.to_string()
    }
}
